src/merkle/simple_smt/tests.rs

```diff
@@ -218,13 +226,16 @@ fn small_tree_opening_is_consistent() {
     ];
 
     for (depth, key, path) in cases {
-        let opening = tree.get_path(NodeIndex::new(depth, key)).unwrap();
+        let opening = tree.get_path(NodeIndex::new(depth, key)?).unwrap();
 
         assert_eq!(path, *opening);
     }
+
+    Ok(())
 }
 
 proptest! {
+    #[cfg(feature = "std")]
```

Comment:
Why test this only for `std`? We should probably keep the PR focused on the proposed change and make a separated one for tests refactor.